### DISCUSSION TITLE
Challenges and Help - Inability to change a scheduled events parameters

### DISCUSSION THREAD

**Discussion by painimation** [StanfordSpezi/.github#64] [Challenges and Help] [answered]
### What Stanford Spezi module is your challenge related to?

SpeziScheduler

### Description

I am not sure how to change a scheduled event or add more events to be scheduled after the app is installed. Could someone help with that? For example, how do we discontinue an event originally scheduled to run for one year, but need to be discontinued after one week? How do we add new events?

### Reproduction

Use the SpeziTemplateApplication
Social support questionnaire is scheduled
Add a new event in TemplateApplicationScheduler
Only the Social support questionnaire is still showing and not the new questionnaire added to the TemplateApplicationScheduler.

### Expected behavior

I would expect that both events added in the TemplateApplicationScheduler to be firing and sending notifications.

### Additional context

There may be a simple way to do this. I just have not figured it out.

### Code of Conduct

- [X] I agree to follow this projects's Code of Conduct and Contributing Guidelines

**Comment by bauer-andreas** [comment, marked as answer]
Hey @painimation,

thank you for creating this discussion and raising the challenges you experienced with the SpeziScheduler module. The shortcoming you described is something we saw as well and is the reason why we invested some resources to rethink some of the SpeziScheduler concepts from the ground up.

We are about to launch a new version of SpeziScheduler (the upcoming 1.0 release) that at its core embraces a versioned, append-only store for Tasks allowing to easily change the contents of a task over its lifetime. This work is incorporated through https://github.com/StanfordSpezi/SpeziScheduler/pull/44. Additionally, the spent some time to improve the way how we automatically schedule notifications for events (if enabled) in https://github.com/StanfordSpezi/SpeziScheduler/pull/49.

Important to note is that the new version won't be compatible with the previous versions (< 1.0.0) of SpeziScheduler. While we make sure pending notification requests are properly removed, previously created tasks and events won't be available.

Hope this helps and we are looking forward what you think about the improvements we made to SpeziScheduler.

Best,
Andreas

**Reply by painimation** [reply]
@Supereg, Many thanks to you and your team for this. This was absolutely needed. Thank you.
